bach/bach/dataframe.py

```diff
@@ -2624,6 +2626,155 @@ def dropna(
+           - "bfill"/"backfill": Fill missing values with the next non-nullable value in the series.
+        :param axis: only ``axis=0`` is supported.
+        :param sort_by: series label or sequence of labels used to sort values.
+            Sorting of values is needed since result might be non-deterministic, as rows with NULLs might
+            yield different results affecting the values to be propagated when using a filling method.
+        :param ascending: Whether to sort ascending (True) or descending (False). If this is a list, then the
+            `by` must also be a list and ``len(ascending) == len(by)``.
+
+        :return: a new dataframe with filled missing values.
+
+        .. note::
+            sort_by is required if DataFrame has no order_by.
+        """
+        df = self.copy()
+        if method and value is not None:
+            raise ValueError('cannot specify both "method" and "value".')
+
+        if method and method not in ('ffill', 'pad', 'bfill', 'backfill'):
+            raise ValueError(f'"{method}" is not a valid method.')
+
+        if method in ('ffill', 'pad'):
+            df = df.ffill(sort_by=sort_by, ascending=ascending)
+
+        elif method in ('bfill', 'backfill'):
+            df = df.bfill(sort_by=sort_by, ascending=ascending)
+
+        if value is not None:
+            series_to_fill = list(value.keys()) if isinstance(value, dict) else self.data_columns
+            for s in series_to_fill:
+                fill_with = value[s] if isinstance(value, dict) else value
+                df[s] = df.all_series[s].fillna(fill_with)
+
+        return df
+
+    def ffill(
+        self,
+        sort_by: Optional[Union[str, Sequence[str]]] = None,
+        ascending: Union[bool, List[bool]] = True,
+    ) -> 'DataFrame':
+        """
+        Fill missing values by propagating the last non-nullable value in each series.
+
+        :param sort_by: series label or sequence of labels used to sort values.
+            Sorting of values is needed since result might be non-deterministic, as rows with NULLs might
+            yield different results affecting the values to be propagated when using a filling method.
+        :param ascending: Whether to sort ascending (True) or descending (False). If this is a list, then the
+            `by` must also be a list and ``len(ascending) == len(by)``.
+
+        :return: a new dataframe with filled missing values.
+
+        .. note::
+            sort_by is required if DataFrame has no order_by.
+        """
+        from bach.partitioning import Window, WindowFrameMode
+        from bach.series.series_numeric import SeriesInt64
+
+        df = self.copy()
+
+        if sort_by:
+            df = df.sort_values(by=sort_by, ascending=ascending)
+
+        if not sort_by and not df.order_by:
+            raise Exception('dataframe must be sorted in order to apply forward fill.')
```

Comment:
```suggestion
            raise Exception('dataframe must be sorted in order to apply forward or backward fill.')
````

This exception can also be indirectly triggered through `bfill()`